### PR TITLE
ID-1264 support role and managed group summary api

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: https://broadworkbench.atlassian.net/browse/ID-1264

What:

these are the sam changes outlined in [this doc](https://docs.google.com/document/d/1tA-Y1mlZlIMEf2Q_Hxao8MiWw2tXBUMYsHpHB8OmjG4/edit#heading=h.w002iup2os6t). Specifically

1. adding a new resource type admin action `admin_read_summary_information`
2. adding a new resource type admin role `support`
3. adding an admin api to check actions on resource type admin
4. adding an admin api to get support summary information for a group

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -424,6 +424,39 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorReport'
+  /api/admin/v1/resourceTypes/{resourceTypeName}/action/{action}:
```
Comment (tlangs) [inline src/main/resources/swagger/api-docs.yaml:427]: Why do we need this? Do we not assume that any admin could get a group summary?

Comment (dvoet) [reply src/main/resources/swagger/api-docs.yaml:427]: This is for other services to call to check if the caller has the `admin_read_summary_information` action. See the [Sam Changes section in the design doc](https://docs.google.com/document/d/1tA-Y1mlZlIMEf2Q_Hxao8MiWw2tXBUMYsHpHB8OmjG4/edit#heading=h.gkaooj4l5b8z) for discussion.

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala [hunk 2]
```diff
@@ -187,36 +192,72 @@ trait AdminRoutes extends SecurityDirectives with SamRequestContextDirectives wi
...
 
-  def adminResourceTypesRoutes(user: SamUser, samRequestContext: SamRequestContext): server.Route =
-    pathPrefix("resourceTypes" / Segment / "policies") { resourceTypeNameToAdminister =>
-      asSamSuperAdmin(user) {
-        withNonAdminResourceType(ResourceTypeName(resourceTypeNameToAdminister)) { resourceTypeToAdminister =>
-          val resource = FullyQualifiedResourceId(resourceTypeAdminName, ResourceId(resourceTypeToAdminister.name.value))
-          pathEndOrSingleSlash {
-            getWithTelemetry(samRequestContext, resourceTypeParam(resourceTypeToAdminister)) {
+  def adminGroupsRoutes(user: SamUser, samRequestContext: SamRequestContext): server.Route =
+    path("groups" / Segment / "supportSummary") { groupName =>
```
Comment (Ghost-in-a-Jar) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala:197]: I think this all needs to be wrapped in `asWorkbenchAdmin` like in `adminUserRoutes`? We really should move up `asWorkbenchAdmin` a level into `adminRoutes` to make it more obvious what is covered by it.

Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala:197]: This is not intended to be `asWorkbenchAdmin`. It checks the required action internally.

Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala:197]: Note that `asWorkbenchAdmin` is an older way of checking for admin access. It requires that the user be in the fc-admins google group. The intent for this api is not to use that google group and instead use a sam action granted on `resource_type_admin`. This is similar to `/api/admin/v1/resources/*` end points.

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala [hunk 3]
```diff
@@ -187,36 +192,72 @@ trait AdminRoutes extends SecurityDirectives with SamRequestContextDirectives wi
...
-          } ~
-          pathPrefix(Segment) { policyName =>
-            val policyId = FullyQualifiedPolicyId(resource, AccessPolicyName(policyName))
+          }
+        }
+      }
+    }
+
+  def adminResourceTypesRoutes(user: SamUser, samRequestContext: SamRequestContext): server.Route =
+    pathPrefix("resourceTypes" / Segment) { resourceTypeNameToAdminister =>
```
Comment (Ghost-in-a-Jar) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala:216]: Similar comment to above do we need to wrap this new endpoint in `asWorkbenchAdmin`? Or is the idea that the user making the call just needs to be on the broad network and have the appropriate role?

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala [hunk 4]
```diff
@@ -238,6 +239,12 @@ object BasicWorkbenchGroup {
 @Lenses final case class ManagedGroupAndRole(groupName: WorkbenchGroupName, role: MangedGroupRoleName)
 @Lenses final case class ManagedGroupMembershipEntry(groupName: ResourceId, role: ResourceRoleName, groupEmail: WorkbenchEmail)
 @Lenses final case class ManagedGroupAccessInstructions(value: String) extends ValueObject
+@Lenses final case class ManagedGroupSupportSummary(
+    groupName: WorkbenchGroupName,
+    email: WorkbenchEmail,
+    authDomainFor: Set[FullyQualifiedResourceId],
+    parentGroups: Set[WorkbenchGroupName]
+)
```
Comment (Ghost-in-a-Jar) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala]: We are starting to try to break out parts of this file into smaller ones, maybe it is worth making a new ManagedGroup file under `model/api`?

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/model/api/SamJsonSupport.scala [hunk 5]
```diff
@@ -111,6 +112,8 @@ object SamJsonSupport {
 
   implicit val ManagedGroupAccessInstructionsFormat = ValueObjectFormat(ManagedGroupAccessInstructions.apply)
 
+  implicit val ManagedGroupSupportSummaryFormat = jsonFormat4(ManagedGroupSupportSummary.apply)
```
Comment (Ghost-in-a-Jar) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/model/api/SamJsonSupport.scala]: If you end up making a ManagedGroup file for the model, the json formatting could go in there as well in the companion object.

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala (hunk 3 above)
Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/api/AdminRoutes.scala:216]: just needs to be on the broad network and have the appropriate role

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala (hunk 4 above)
Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala]: I don't see that anyone has started on this approach yet and I don't really want to be the first

Comment (tlangs) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala]: https://github.com/broadinstitute/sam/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/sam/model/api/SamUser.scala

**Comment by dvoet** [comment]
I moved the managed group model classes a json formatters to src/main/scala/org/broadinstitute/dsde/workbench/sam/model/api/ManagedGroupModel.scala

**Review by Ghost-in-a-Jar — APPROVED** [review]

**Review by tlangs — APPROVED** [review]

**Pull request merged**